stackslib/src/net/mod.rs

```diff
@@ -836,19 +836,47 @@ pub struct Preamble {
 
 /// Request for a block inventory or a list of blocks.
 /// Aligned to a PoX reward cycle.
+/// This struct is used only in Stacks 2.x for Stacks 2.x inventories
 #[derive(Debug, Clone, PartialEq)]
 pub struct GetBlocksInv {
-    pub consensus_hash: ConsensusHash, // consensus hash at the start of the reward cycle
-    pub num_blocks: u16,               // number of blocks to ask for
+    /// Consensus hash at thestart of the reward cycle
+    pub consensus_hash: ConsensusHash,
+    /// Number of sortitions to ask for. Can be up to the reward cycle length.
+    pub num_blocks: u16,
 }
 
 /// A bit vector that describes which block and microblock data node has data for in a given burn
-/// chain block range.  Sent in reply to a GetBlocksInv.
+/// chain block range.  Sent in reply to a GetBlocksInv for Stacks 2.x block data.
 #[derive(Debug, Clone, PartialEq)]
 pub struct BlocksInvData {
-    pub bitlen: u16, // number of bits represented in bitvec (not to exceed PoX reward cycle length).  Bits correspond to sortitions on the canonical burn chain fork.
-    pub block_bitvec: Vec<u8>, // bitmap of which blocks the peer has, in sortition order.  block_bitvec[i] & (1 << j) != 0 means that this peer has the block for sortition 8*i + j
-    pub microblocks_bitvec: Vec<u8>, // bitmap of which confirmed micrblocks the peer has, in sortition order.  microblocks_bitvec[i] & (1 << j) != 0 means that this peer has the microblocks produced by sortition 8*i + j
+    /// Number of bits in the block bit vector (not to exceed the reward cycle length)
+    pub bitlen: u16,
+    /// The block bitvector. block_bitvec[i] & (1 << j) != 0 means that this peer has the block for
+    /// sortition 8*i + j.
+    pub block_bitvec: Vec<u8>,
+    /// The microblock bitvector. microblocks_bitvec[i] & (1 << j) != 0 means that this peer has
+    /// the microblocks for sortition 8*i + j
+    pub microblocks_bitvec: Vec<u8>,
+}
+
+/// Request for a tenure inventroy.
+/// Aligned to a PoX reward cycle.
+/// This struct is used only in Nakamoto, for Nakamoto inventories
+#[derive(Debug, Clone, PartialEq)]
+pub struct GetNakamotoInvData {
+    /// Consensus hash at the start of the reward cycle
+    pub consensus_hash: ConsensusHash,
+}
+
+/// A bit vector that describes Nakamoto tenure availability.  Sent in reply for GetBlocksInv for
+/// Nakamoto block data.
```

Comment:
My understanding from reading the generator code is:

`inv[i] = true` means that the i-th bitcoin block in a given reward cycle has a tenure start in the given block _and_ that this node has processed all the blocks of that sortition.

`inv[i] = false` is the inverse of that (i.e., either no tenure started at the block or this node has not processed all the blocks of that sortition).

Is that understanding correct? If so, can you add it to this rustdoc?

Reply:
Correct -- if the node didn't process a tenure's data, then the bit for it is 0 (note that this is different from epoch 2, where you'd get a 1 as long as the node had a copy of the block, even if it was not processed).  I'll update the doc.